lib/ansible/modules/network/ovs/openvswitch_bridge.py

```diff
@@ -123,12 +125,23 @@ def map_obj_to_commands(want, have, module):
             commands.append(command)
     else:
         if have:
-            if want['fail_mode'] != have['fail_mode']:
-                templatized_command = ("%(ovs-vsctl)s -t %(timeout)s"
-                                       " set-fail-mode %(bridge)s"
-                                       " %(fail_mode)s")
-                command = templatized_command % module.params
-                commands.append(command)
+            wanted_fail_mode = want['fail_mode']
```

Comment:
@ashish-billore  It looks much better now. Thanks for making changes. One comment here is passing empty value to ansible module arg does not look good. Probably we can pass something as 'clear' to convey user intention to clear previously set fail_mode. Also fail_mode is a choice parameter in doc string so it has to be a choice parameter in code as well. This also need fixing. I could test following code to be working as expected. please take a look
```
--- a/lib/ansible/modules/network/ovs/openvswitch_bridge.py
+++ b/lib/ansible/modules/network/ovs/openvswitch_bridge.py
@@ -53,9 +53,11 @@ options:
               To  clear all external-ids pass an empty value.
     fail_mode:
         version_added: 2.0
-        choices : [secure, standalone]
+        choices : [secure, standalone, clear]
         description:
             - Set bridge fail-mode. The default value (None) is a No-op.
+              Omitting this parameter is a No-op.
+              To clear previously set fail_mode, pass it as 'clear'.
     set:
         version_added: 2.3
         description:
@@ -123,12 +125,23 @@ def map_obj_to_commands(want, have, module):
             commands.append(command)
     else:
         if have:
-            if want['fail_mode'] != have['fail_mode']:
-                templatized_command = ("%(ovs-vsctl)s -t %(timeout)s"
-                                       " set-fail-mode %(bridge)s"
-                                       " %(fail_mode)s")
-                command = templatized_command % module.params
-                commands.append(command)
+            wanted_fail_mode = want['fail_mode']
+            have_fail_mode = have['fail_mode']
+            if wanted_fail_mode != have_fail_mode:
+                if wanted_fail_mode is not None:
+                    if wanted_fail_mode != 'clear':
+                        # only set the fail-mode if specified and not empty
+                        templatized_command = ("%(ovs-vsctl)s -t %(timeout)s"
+                                               " set-fail-mode %(bridge)s"
+                                               " %(fail_mode)s")
+                    else:
+                        # if fail-mode is specified, but empty it means user
+                        # wants to reset the fail-mode, so clear it off
+                        templatized_command = ("%(ovs-vsctl)s -t %(timeout)s"
+                                               " del-fail-mode %(bridge)s")
+
+                    command = templatized_command % module.params
+                    commands.append(command)
 
             if want['external_ids'] != have['external_ids']:
                 templatized_command = ("%(ovs-vsctl)s -t %(timeout)s"
@@ -233,7 +246,8 @@ def main():
         'state': {'default': 'present', 'choices': ['present', 'absent']},
         'timeout': {'default': 5, 'type': 'int'},
         'external_ids': {'default': None, 'type': 'dict'},
-        'fail_mode': {'default': None},
+        'fail_mode': {'default': None, 'choices': ['secure', 'standalone',
+            'clear']},
         'set': {'required': False, 'default': None}
     }
 
```

Reply:
thanks @gdpak looks good to me, minor updates to the comment in the above else check for clear:
```suggestion
 +                    else:
+                        # if fail-mode is specified as 'clear' means user
+                        # wants to reset the fail-mode, so clear it off
```